src/platform/nrfconnect/FactoryDataParser.c

```diff
@@ -0,0 +1,160 @@
+/*
```

Comment:
Any reason for this file being c instead of cpp?

Otherwise we need extern C in the header or linkage will be wrong.

Reply:
Yes, the reason is that `zcbor_decode.h` library that is used needs to be compiled with C compiler and not C++, otherwise compilation with fail due to invalid syntax that is used from C++ perspective (taking address of rvalue [-fpermissive]).